userbenchmark/ddp_experiments/parse_ddp.py

```diff
@@ -8,47 +10,32 @@
 import tabulate
 import sys
 
-def get_job_result(args, job_id, worker_rank=0):
-    root = os.path.join(args.results_dir, f"{job_id}_{worker_rank}_")
-    err = root + "log.err"
-    out = root + "log.out"
-    pkl = root + "result.pkl"
-    if not os.path.isfile(pkl):
-        return False, f"waiting.. or {pkl} did not exist"
-
-    with open(pkl, 'rb') as f:
-        dat = pickle.load(f)
-        assert isinstance(dat, tuple), f"Expected a tuple as result but got a {type(dat)}: {dat}"
-        desc, payload = dat
-        if desc == "error":
-            # print(f"Got error: {desc}, traceback:\n{payload}")
-            return False, payload
-        elif desc == "success":
-            # print(f"Success: {payload}")
-            return True, payload
-
-        # print(f"Unknown result: {dat}")
-        return False, dat
-
-    return False, None
-
-def parse_data(args):
+def parse_partial(args):
     """
     Schema:
-    model_data["model"]["backend"][#nodes] = latency_median
+    model_data["model"]["backend"][#nodes] = result
+    where "result" can be a list of results, or "error"
     """
-    model_data = defaultdict(lambda: defaultdict(lambda: defaultdict(lambda: defaultdict(float))))
-    with open(args.csv) as f:
-        runs = csv.DictReader(f)
-        for row in runs:
-            model = row["model"]
+    model_data = defaultdict(lambda: defaultdict(lambda: defaultdict(lambda: defaultdict(list))))
+    rank_id = 0
+    log_path = os.path.join(args.results_dir, f"{args.job_id}_{rank_id}_log.out")
+    with open(log_path, "r") as f:
+        content = f.read()
+        pieces = content.split("<RESULT>")
```

Comment:
How is the log file format defined? Will it be subjected to change and break this code?

Reply:
The log format is defined in ddp_experiments.py, it's just the `print(f"<RESULT>...` statement. I can follow up with a PR that uses the pkl file by default and falls back to this only if the pkl file isn't available.